lib/spack/spack/test/stage.py

```diff
@@ -177,82 +194,32 @@ def get_stage_path(stage, stage_name):
 
 
 @pytest.fixture
-def bad_stage_path():
-    """Temporarily ensure there is no accessible path for staging."""
-    current = spack.config.get('config:build_stage')
-    spack.config.set('config', {'build_stage': '/no/such/path'}, scope='user')
-    yield
-    spack.config.set('config', {'build_stage': current}, scope='user')
-
-
-@pytest.fixture
-def non_user_path_for_stage(monkeypatch):
-    """Temporarily use a Linux-standard non-user path for staging. """
-    def _can_access(path, perms):
-        return True
-
-    current = spack.config.get('config:build_stage')
-    spack.config.set('config', {'build_stage': [_non_user_root]}, scope='user')
-    monkeypatch.setattr(os, 'access', _can_access)
-    yield
-    spack.config.set('config', {'build_stage': current}, scope='user')
-
-
-@pytest.fixture
-def instance_path_for_stage():
-    """
-    Temporarily use the "traditional" spack instance stage path for staging.
-
-    Note that it can be important for other tests that the previous settings be
-    restored when the test case is over.
-    """
-    current = spack.config.get('config:build_stage')
+@pytest.mark.nomockstage
+def instance_path_for_stage(clear_stage_root):
+    """Use the spack instance stage path for the stage root."""
     base = canonicalize_path(os.path.join('$spack', 'test-stage'))
```

Comment:
Shouldn't this be done with a tmpdir feature so we don't pollute the real file system?

Reply:
This test is for coverage purposes related to using an instance directory.  The alternative would be to not include the associated test.

I am minimizing the impact on the real system by using a test fixture (that ensures filesystem cleanup) and `spack.config.overrides` (to ensure the configuration file is not polluted).

Reply:
This looks like a fixture vs. a test. I only see this being used in `test_get_stage_root_in_spack`. I think that is testing that you can set `_stage_root` via overriding the config variable - am I right about that? If so I think the naming and distribution of logic between this fixture and the test using it is confusing.

Reply:
I don't think that's quite right.  This test was intended to confirm that `config:build_stage` using an instance path properly sets the stage root.  (I should also be checking that the directory exists.)

But I think I see your point.  Would it help if I add a check that the directory exists?

Reply:
>  This test was intended to confirm that config:build_stage using an instance path properly sets the stage root

To check: what is an "instance path"?

Reply:
Spack instance (using terminology in the docs).

Reply:
Added check to ensure the directory exists.

Reply:
If you are testing that:

```
spack.config.override('config:build_stage',  test_path)
```

sets `_stage_root`, I think that this line should appear in `test_get_stage_root_in_spack`: the code/behavior you want to test should appear in the test and not the fixture. The fixture should be used to mock any components of the system that need to be mocked for the duration of the test; that's not to say that all setup code must appear in fixtures, but rather if there is a function call and you want to make sure it works, it should not appear in the fixture.

Reply:
You mean as a comment?

Reply:
No I mean that the fixture should be merged into the test (assuming I understand correctly that the point of the test is to make sure `spack.config.override('config:build_stage'...)` works).

Reply:
Done.